crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -1021,12 +1021,14 @@ where
                     }
                 }
             }
-            Stmt::ImportFrom(ast::StmtImportFrom {
-                names,
-                module,
-                level,
-                range: _,
-            }) => {
+            Stmt::ImportFrom(
+                imp_from @ ast::StmtImportFrom {
```

Comment:
@MichaReiser I will start doing this for any rules I implement now, as you have recommended it a couple times now, and I like the cleanliness of not having to break down all the required arguments for any given rule func. 

Just for my understanding/education. Is this a new feature that Rust/Ruff supports that was not there before, and that's why almost all other funcs expect a selection of structfields as arguments? Or is this just a preference thing that other devs have decided otherwise on?

If the former is true, would it be worthwhile to refactor all existing rules to accept a whole Stmt and pick the required fields in the rule func? If so I could create an Issue for it and maybe look at doing that depending on how much work that would be.

Reply:
Thanks for the clear explanation! Will continue using the `ast::` kinds for future plugins and wait with anything else until there's news on the API change.

Reply:
We'll desperately need help with a migration once we've refined the API 😂